homeassistant/components/sensibo/climate.py

```diff
@@ -138,208 +137,234 @@ async def async_assume_state(service: ServiceCall) -> None:
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the fan setting."""
+        return self.coordinator.data[self.unique_id]["fan_mode"]
+
+    @property
+    def fan_modes(self) -> list[str] | None:
+        """Return the list of available fan modes."""
+        return self.coordinator.data[self.unique_id]["fan_modes"]
+
+    @property
+    def swing_mode(self) -> str | None:
+        """Return the swing setting."""
+        return self.coordinator.data[self.unique_id]["swing_mode"]
+
+    @property
+    def swing_modes(self) -> list[str] | None:
+        """Return the list of available swing modes."""
+        return self.coordinator.data[self.unique_id]["swing_modes"]
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return self.coordinator.data[self.unique_id]["temp_list"][0]
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return self.coordinator.data[self.unique_id]["temp_list"][-1]
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self.coordinator.data[self.unique_id]["available"] and super().available
 
     async def async_set_temperature(self, **kwargs) -> None:
         """Set new target temperature."""
         if (temperature := kwargs.get(ATTR_TEMPERATURE)) is None:
             return
-        temperature = int(temperature)
-        if temperature not in self._temperatures_list:
+
+        if temperature == self.target_temperature:
+            return
+
+        if temperature not in self.coordinator.data[self.unique_id]["temp_list"]:
             # Requested temperature is not supported.
-            if temperature == self.target_temperature:
-                return
-            index = self._temperatures_list.index(self.target_temperature)
-            if (
-                temperature > self.target_temperature
-                and index < len(self._temperatures_list) - 1
-            ):
-                temperature = self._temperatures_list[index + 1]
-            elif temperature < self.target_temperature and index > 0:
-                temperature = self._temperatures_list[index - 1]
+            if temperature > self.coordinator.data[self.unique_id]["temp_list"][-1]:
+                temperature = self.coordinator.data[self.unique_id]["temp_list"][-1]
+
+            elif temperature < self.coordinator.data[self.unique_id]["temp_list"][0]:
+                temperature = self.coordinator.data[self.unique_id]["temp_list"][0]
+
             else:
                 return
 
-        await self._async_set_ac_state_property("targetTemperature", temperature)
+        result = await self._async_set_ac_state_property(
+            "targetTemperature", int(temperature)
+        )
+        if result:
+            self.coordinator.data[self.unique_id]["target_temp"] = int(temperature)
+            self.async_write_ha_state()
```

Comment:
This section of calling `_async_set_ac_state_property`, checking the result and setting the updated value in the `coordinator` repeat itself few times, I suggest either to modify the `_async_set_ac_state_property` to set the correct value in the coordinator upon success or use a wrapper function, for example:
https://github.com/home-assistant/core/blob/dev/homeassistant/components/webostv/media_player.py#L103-L121